worker/caasoperator/caasoperator.go

```diff
@@ -429,7 +429,7 @@ func charmModified(local LocalState, remote remotestate.Snapshot) bool {
 
 func (op *caasOperator) ensureCharm() (*charm.URL, int, error) {
 	charmDir := op.paths.GetCharmDir()
-	if _, err := os.Stat(charmDir); !os.IsNotExist(err) {
+	if _, err := os.Stat(charmDir); err != nil && !os.IsNotExist(err) {
```

Comment:
Do you really need the `err != nil` here - doesn't `os.IsNotExist` return false for nil? I tried it in the playground and it worked.

Reply:
!os.IsNotExist(err) returns true for nil err and thus returns without doing the download. and worse, the returned error is nil and so the caller thinks there's nothing wrong but actually nothing was downloaded

Reply:
Ok, I get this now you explained it in IRC.

Reply:
Actually,the original code was correct after all. The operator was too simplistic in how it determined if a charm already existed. I needed to adopt the code already used in the IAAS uniter.